pkg/roachpb/metadata.go

```diff
@@ -946,7 +956,53 @@ func (h *GCHint) ForwardLatestRangeDeleteTimestamp(ts hlc.Timestamp) bool {
 	return false
 }
 
-// ResetLatestRangeDeleteTimestamp resets delete range timestamp.
-func (h *GCHint) ResetLatestRangeDeleteTimestamp() {
-	h.LatestRangeDeleteTimestamp = hlc.Timestamp{}
+// ScheduleGCFor updates the hint to schedule GC for the given timestamp.
+// Returns true iff the hint was updated.
+func (h *GCHint) ScheduleGCFor(ts hlc.Timestamp) bool {
```

Comment:
Maybe this is excessive, but is it worth passing in `now` here and clamping the timestamp to it, to guard against future timestamps? If this happened, it would ruin the 2xTTL worst-case guarantee. Then again, we can probably assume these timestamps to have interacted with the HLC, and if we ever got future timestamps into the HLC we're pretty screwed anyway. Probably safe to drop, just leaving it for your consideration.

Reply:
Meh, let's drop this -- if we have far-future HLC timestamps then we have much bigger problems than slow GC.